crates/lox_core/src/time/continuous.rs

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (c) 2023. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::fmt;
+use std::fmt::Formatter;
+use std::ops::{Add, Sub};
+
+use num::ToPrimitive;
+
+use crate::time::constants::i64::{SECONDS_PER_DAY, SECONDS_PER_HOUR, SECONDS_PER_MINUTE};
+use crate::time::constants::u64::ATTOSECONDS_PER_SECOND;
+use crate::time::dates::Calendar::ProlepticJulian;
+use crate::time::dates::Date;
+use crate::time::scales::TimeScale;
+use crate::time::utc::{UTCDateTime, UTC};
+use crate::time::{constants, Thousandths, WallClock};
+
+/// `ContinuousTime` is the base time representation for time scales without leap seconds. It is measured relative to an
+/// arbitrary epoch. `RawTime::default()` represents the epoch itself.
+///
+/// `ContinuousTime` has attosecond precision, and supports times within 292 billion years either side of the epoch. The
+/// sign of the time is determined by the sign of the `second` component.
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+pub struct ContinuousTime {
+    pub seconds: i64,
+    pub attoseconds: u64,
+}
+
+impl Add<Delta> for ContinuousTime {
+    type Output = Self;
+
+    fn add(self, rhs: Delta) -> Self::Output {
+        let mut attoseconds = self.attoseconds + rhs.attoseconds;
+        let mut seconds = self.seconds + rhs.seconds as i64;
+        if attoseconds >= ATTOSECONDS_PER_SECOND {
+            seconds += 1;
+            attoseconds -= ATTOSECONDS_PER_SECOND;
+        }
+        Self {
+            seconds,
+            attoseconds,
+        }
+    }
+}
+
+impl Sub<Delta> for ContinuousTime {
+    type Output = Self;
+
+    fn sub(self, rhs: Delta) -> Self::Output {
+        let mut seconds = self.seconds - rhs.seconds as i64;
+        let mut attoseconds = self.attoseconds;
+        if rhs.attoseconds > self.attoseconds {
+            seconds -= 1;
+            attoseconds = ATTOSECONDS_PER_SECOND - (rhs.attoseconds - self.attoseconds);
+        } else {
+            attoseconds -= rhs.attoseconds;
+        }
+        Self {
+            seconds,
+            attoseconds,
+        }
+    }
+}
+
+/// An absolute continuous time difference with attosecond precision.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct Delta {
```

Comment:
I would prefer `TimeDelta` (because `astropy` uses the same terminology) but it's not a hill I am willing to do on.